### PR TITLE
Readme refactor & Add Appache License 

### PR DESCRIPTION
**Pull request by sekulicd**
This refactors Readme file and adds Apache license file.

@tiero @richiejp please review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,167 +1,29 @@
-# Prem Operator
+## Prem Operator: Advanced AI Model Deployment on Kubernetes
 
-Deploy AI models to Kubernetes using LocalAI, vLLM, DeepSpeed-MII, NVIDIA Triton and custom Docker images.
```
Comment (richiejp) [inline README.md:3]: I'd like to keep a one line description which specifies what the operator does with the minimum number of words. We could maybe add "easily" at the start?

File: docs/deployment.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Deployment
+
+## Requirements
+
+- K8s cluster
+- Helm
+- Ingress controller (e.g. Traefik)
+- Nvidia GPU Operator
```
Comment (richiejp) [inline docs/deployment.md:8]: The last three are recommendations. You can install without Helm, you don't need ingress if you are serving the model internally and you can use CPU for embeddings.

File: docs/deployment.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+## Artifacts
+Prem-Operator has three artifacts:
+1. [Source Code](https://github.com/premAI-io/saas-controller)
+1. [Prem-Operator Docker image](#container-images)
+1. [Prem-Operator Helm chart](https://hub.docker.com/r/premai/prem-operator-chart)
+
+## Installation
+After setting up K8s cluster, install [Nvidia GPU Operator](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/getting-started.html) and [Traefik](https://doc.traefik.io/traefik/getting-started/install-traefik/#use-the-helm-chart) as an ingress controller.
+Note that Nvidia GPU Operator is required for GPU support and Traefik is required for handling ingress traffic.
```
Comment (richiejp) [inline docs/deployment.md:18]: These are recommendations, but it is good to point out this is what we have tested it with

File: docs/developer_guide.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+After deploying you can infer the model using the following curl command:
+```
+curl http://foo.127.0.0.1.nip.io:8080/v1/completions -H "Content-Type: application/json" -d '{
+"model": "tinyllama-chat",
+"prompt": "Do you always repeat yourself?",
+"temperature": 0.1,
+"max_tokens": 50
+}'
+```
```
Comment (richiejp) [inline docs/developer_guide.md:117]: this file is all good

File: docs/contributing.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+## Making Contributions
+
+### Bug Fixes
+
+- If you've found a bug and have a fix, simply **open a Pull Request (PR)** with your changes.
+- Ensure your PR description clearly describes the bug and fix. Include any relevant issue numbers.
+
+### New Features
+
+- For new features, we ask that you **open an issue first** to discuss your ideas with the project maintainers.
```
Comment (richiejp) [inline docs/contributing.md:17]: I don't mind if people make a PR first. I think the main thing is that they understand that without buy-in it's possible that we won't review the code, but just dismiss it on principle.

File: README.md (hunk 1 above)
Comment (sekulicd) [reply README.md:3]: Can you suggest exact sentence you would like to see?

Comment (richiejp) [reply README.md:3]: That sentence with "Easily" on the front would be fine IMO

File: docs/deployment.md (hunk 3 above)
Comment (richiejp) [reply docs/deployment.md:18]: For instance add "optionally", before install and change "Traefik is required" to "Traefik is on option"

File: README.md (hunk 1 above)
Comment (sekulicd) [reply README.md:3]: Done

File: docs/deployment.md (hunk 2 above)
Comment (sekulicd) [reply docs/deployment.md:10]: Done

File: docs/deployment.md (hunk 3 above)
Comment (sekulicd) [reply docs/deployment.md]: Done

File: docs/contributing.md (hunk 5 above)
Comment (sekulicd) [reply docs/contributing.md]: Done

**Review by richiejp — APPROVED** [review]
Great work.

**Pull request merged**